app/uk/gov/hmrc/merchandiseinbaggage/views/PreviousDeclarationDetailsView.scala.html

```diff
@@ -0,0 +1,103 @@
+@*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.govukfrontend.views.html.components
+@import uk.gov.hmrc.merchandiseinbaggage.model.api.DeclarationType
+@import uk.gov.hmrc.merchandiseinbaggage.views.html.Layout
+@import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig
+@import uk.gov.hmrc.merchandiseinbaggage.controllers.routes._
+@import uk.gov.hmrc.merchandiseinbaggage.model.api.ImportGoods
+@import uk.gov.hmrc.merchandiseinbaggage.model.api.ExportGoods
+@import uk.gov.hmrc.merchandiseinbaggage.controllers.routes._
+@import uk.gov.hmrc.merchandiseinbaggage.model.api.Goods
+@import uk.gov.hmrc.merchandiseinbaggage.model.api.JourneyDetails
+@import uk.gov.hmrc.merchandiseinbaggage.utils.DateUtils._
+
+@this(
+        layout: Layout,
+        formHelper: FormWithCSRF,
+        h1: components.h1,
+        h2: components.h2,
+        p: components.p,
+        link: components.link,
+        button: components.button,
+)
+
+@(goods: Seq[Goods], journeyDetails: JourneyDetails,  declarationType: DeclarationType, withinTimerange:Boolean )(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+
+@layout(pageTitle = Some(titleNoForm(messages("previousDeclarationDetails.title"))), maybeBackButtonUrl = Some(RetrieveDeclarationController.onPageLoad), maybeDeclarationType = None) {
+    @h1(("previousDeclarationDetails.heading"))
+
+    @h2(("previousDeclarationDetails.details"), Some("govuk-heading-m"))
+
+    @for((good, index) <- goods.zipWithIndex) {
+        <dl class="govuk-summary-list govuk-!-margin-bottom-9">
```

Comment:
It's quite messy to have all this HTML with classes etc in our views. We should be using the govuk components

You can inject GovukSummaryList and construct the list that way. Have a look in the viewModels.GoodsSummary object for an example, you should be able to largely copy and paste from there and get rid of the `actions` as we don't want change links on this page.

Not a major issue, just nicer to look at and easier to maintain in my opinion